documentation/proposals/Proposal - Axis Input Devices.md

```diff
@@ -0,0 +1,624 @@
+}
+```
+
+## IAxisInputHandler
+
+This is how a developer could subscribe directly to the axis changes of an `IAxisDevice`. It mimicks the `IGamepadInputHandler` interface from the [Multi-Backend Input proposal](./Proposal - Multi-Backend Input.md).
+
+```csharp
+
+public interface IAxisInputHandler
+{
+    public void OnAxisChanged(IAxisDevice device, AxisDescription description, float value);
+}
+```
+
+## Axis Descriptions
+
+There are two primary structures that give meaning to the axes of an `IAxisDevice`: `AxisDescription` and `AxisGroup`. The former defines the axes themselves, and the latter defines how these axes relate to one another. Note that both of these structures have an `Index` field, which must match the index of the `IAxisDevice`'s collection that it is defined in.
+
+### Struct Definition
+
+```csharp
+/// <summary>
+/// This describes a single axis on a device
+/// </summary>
+/// <param name="Index">the index of the axis in the device</param>
+/// <param name="Traits">the traints of the axis, described above</param>
+/// <param name="Name">A human-readable name for the axis</param>
+/// <param name="RawValueBounds">The bounds of the raw value - a value of 'default' indicates it is typical (0, 1), or
+/// in the case that the provided traits are <see cref="AxisTrait.RawValueOnly"/>, this must be default and indicates no known bounds.</param>
+public readonly record struct AxisDescription(
+    int Index,
+    AxisTrait Traits,
+    string Name,
+    Vector2 RawValueBounds = default)
+{
+    public static implicit operator int(AxisDescription description) => description.Index;
+}
+```
+
+### Axis Traits
+These are the primary flags through which an axis's use is defined. Its definition includes several constraints meant to both make defining axes less cumbersome and enforce logically consistent behaviour where possible. This enum is one of the primary places where validation needs to occur at runtime, and some flags place implementation requirements on the implementer. Wherever possible, we should seek to provide default implementations of expected behaviours regarding these constraints.
+
+```csharp
+
+/// <summary>
+/// This is how individual axes are defined
+/// Pay attention to the flags which require specific implementation details related to their RawValues and normalization
+/// </summary>
+[Flags]
+public enum AxisTrait
+{
+    Unknown = 0,
+    Analog = 1 << 0,
+    Binary = 1 << 1,
+
+    // an analog axis that explicitly does not return to center or zero like a joystick or trigger
+    // things like touch-pads, mice, or world-space positions
+    Point = 1 << 8 | Analog | HasRawValue,
+
+    // ---- Physical orientation ----
+    Orientation = 1 << 2 | Analog,
+    Rotation = 1 << 3 | Orientation,
+
+    // this is specified because single axes of euler angle components can be useful by themselves,
+    // whereas anything shorter than 3 or 4 axes is not for quaternion rotation components

```

Comment:
Is the ordering of `AxisDescription.Index` important for grouping multiple rotation components into a quaternion for instance? If so, should be specified in the group requirements. Is azimuth/altitude considered here or does this have to be converted before being reported as an axis? And other similar questions - basically how is the user of the axis device intended to group and interpret rotational components into a transformation they can apply to a 3D model for instance.

Reply:
just a note - azimuth would likely be a Rotation axis group that could be separated from the "gyroscopic" rotation if need be,  and altitude would likely be the Z component of a Position3D axis group, or a single Position1D axis group